zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/NotebookRepoSync.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.notebook.repo;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.conf.ZeppelinConfiguration.ConfVars;
+import org.apache.zeppelin.notebook.Note;
+import org.apache.zeppelin.notebook.NoteInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Notebook repository sync with remote storage
+ */
+public class NotebookRepoSync implements NotebookRepo{
+  private VFSNotebookRepo localRepo;
+  private NotebookRepo remoteRepo;
+  private static final Logger LOG = LoggerFactory.getLogger(NotebookRepoSync.class);
+
+  /**
+   * @param (conf)
+   * @throws - Exception
+   */
+  public NotebookRepoSync(ZeppelinConfiguration conf) throws Exception {
+    localRepo = new VFSNotebookRepo(conf);
+    final String defStorageClassName = "org.apache.zeppelin.notebook.repo.VFSNotebookRepo";
```

Comment:
Hard-coding constants like that is really not a best practice to have maintainable code. What if the name changes later on?

We should find a better place for this value.